android/app/src/main/java/io/mosip/registration_client/api_services/BiometricsDetailsApi.java

```diff
@@ -0,0 +1,553 @@
+    private byte[] byteArrayTester;
+    private List<byte[]> listByteArrayTester1=new ArrayList<>();
+    BiometricsPigeon.Result<String> result1;
+
+    @Inject
+        public BiometricsDetailsApi(AuditManagerService auditManagerService, ObjectMapper objectMapper, Biometrics095Service biometrics095Service, RegistrationService registrationService,GlobalParamRepository globalParamRepository) {
+        this.auditManagerService = auditManagerService;
+        this.objectMapper = objectMapper;
+        this.biometricsService=biometrics095Service;
+        this.registrationService=registrationService;
+        this.globalParamRepository=globalParamRepository;
+
+    }
+
+    public void setCallbackActivity(MainActivity mainActivity){
+        this.activity=mainActivity;
+    }
+
+
+    @Override
+    public void invokeDiscoverSbi(@NonNull String fieldId, @NonNull String modality, @NonNull BiometricsPigeon.Result<String> result) {
+        currentModality = getModality(modality);
+        this.fieldId=fieldId;
+        discoverSBI();
+        result1=result;
+    }
+
+    @Override
+    public void getBestBiometrics(@NonNull String fieldId, @NonNull String modality, @NonNull BiometricsPigeon.Result<List<String>> result) {
+        try{
+
+            RegistrationDto registrationDto=registrationService.getRegistrationDto();
+
+            ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+            String json;
+            List<String> jsonList=new ArrayList<>();
+        biometricsDtoList=registrationDto.getBestBiometrics(fieldId,getModality(modality));
+            for(int i=0;i<biometricsDtoList.size();i++){
+                json=ow.writeValueAsString(biometricsDtoList.get(i));
+                jsonList.add(json);
+
+            }
+
+            result.success(jsonList);
+
+
+        }catch (Exception e){
+            Log.e(TAG,e.getMessage());
+        }
+    }
+
+    @Override
+    public void getBiometrics(@NonNull String fieldId, @NonNull String modality, @NonNull Long attempt, @NonNull BiometricsPigeon.Result<List<String>> result) {
+        try{
+            RegistrationDto registrationDto=registrationService.getRegistrationDto();
+            ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+            String json;
+            List<String> jsonList=new ArrayList<>();
+            biometricsDtoList=registrationDto.getBiometrics(fieldId,getModality(modality),attempt.intValue()-1);
+
+            for(int i=0;i<biometricsDtoList.size();i++){
+                json=ow.writeValueAsString(biometricsDtoList.get(i));
+                jsonList.add(json);
+
+            }
+
+            result.success(jsonList);
+        }catch(Exception e){
+            Log.e(TAG,e.getMessage());
+        }
+    }
+
+    @Override
+    public void extractImageValues(@NonNull String fieldId, @NonNull String modality, @NonNull BiometricsPigeon.Result<List<byte[]>> result) {
```

Comment:
extractImageValues and  extractImageValuesByAttempt are very similar in logic

ideally 

we should separate the logic of getting biovalue and decoding logic so there is no duplication of functions